File: scales/scales-info/scalesInfo.py
# -----------------------------------------------------------------------------
#
# Example for info of cv.Mat type in Python3 with OpenCV.
#
# -----------------------------------------------------------------------------

import cv2

image = cv2.imread("image.png")

if image is None:
    exit(-1)

print("Size:", image.size)
print("Shape:", image.shape)
print("Data type:", image.dtype)
print("Item size:", image.itemsize)
print("Number of dimensions:", image.ndim)
print("Number of channels:", image.shape[2])
print("Number of rows:", image.shape[0])
print("Number of columns:", image.shape[1])
print("Total number of elements:", image.size)
print("Size of each element:", image.itemsize)
print("Size of each element without channels:", image.itemsize // image.shape[2] if len(image.shape) == 3 else image.itemsize)

File: scales/scales-info/scalesInfo.cpp
//-----------------------------------------------------------------------------
//
// Example for info of cv::Mat type in C++ with OpenCV.
//
//-----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

int main() {
    cv::Mat image = cv::imread("image.png");

    if (image.empty()) exit(-1);

    std::cout << "size: " << image.size() << '\n';
    std::cout << "channels: " << image.channels() << '\n';
    std::cout << "depth: " << image.depth() << '\n';
    std::cout << "rows: " << image.rows << '\n';
    std::cout << "cols: " << image.cols << '\n';
    std::cout << "empty: " << image.empty() << '\n';
    std::cout << "type: " << image.type() << '\n';
    std::cout << "total: " << image.total() << '\n';
    std::cout << "elemSize: " << image.elemSize() << '\n';
    std::cout << "elemSize1: " << image.elemSize1() << '\n';
}
